### PR TITLE
Discover oldest input timestamp

### PR DESCRIPTION
**Pull request by fabyan-alexander**

### CODE REVIEW

**Comment by fabyan-alexander** [comment]
[[BillingAg] Identify the oldest day in a folder](https://trello.com/c/PZzqRKhY/965-billingag-identify-the-oldest-day-in-a-folder)

File: src/test/java/com/leaf/billing/FileUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class FileUtilTest {
+
+    @Test
+    void extractTimestampFromFileName() {
+        String name = "boundaryEvent2_2022-01-01T13-00-00.parquet";
+        LocalDate date = com.leaf.billing.utils.FileUtils.extractTimestampFromFileName(name);
+        assertThat(date.get(ChronoField.YEAR)).isEqualTo(2022);
+        assertThat(date.get(ChronoField.MONTH_OF_YEAR)).isEqualTo(1);
+        assertThat(date.get(ChronoField.DAY_OF_MONTH)).isEqualTo(1);
```
Comment (ricardosantacattarina) [inline src/test/java/com/leaf/billing/FileUtilTest.java:18]:
```suggestion
    void extractTimestampFromFileName() {
        String name = "boundaryEvent2_2022-05-01T13-00-00.parquet";
        LocalDate date = com.leaf.billing.utils.FileUtils.extractTimestampFromFileName(name);
        assertThat(date.get(ChronoField.YEAR)).isEqualTo(2022);
        assertThat(date.get(ChronoField.MONTH_OF_YEAR)).isEqualTo(5);
        assertThat(date.get(ChronoField.DAY_OF_MONTH)).isEqualTo(1);
```

I know it probably works, but having the test assert the same value for different properties might not be ideal.

**Review by ricardosantacattarina — APPROVED** [review]
Nice work!

File: src/main/java/com/leaf/billing/utils/SparkCommands.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+@Slf4j
+public class SparkCommands {
+
+    private SparkCommands() {
+        //NO-OP
+    }
+
+    public static LocalDate locateOldestTimestampGroup(String path) throws IOException {
+        var fileSystem = FileSystem.get(URI.create(path), new Configuration());
+        var it = fileSystem.listFiles(new Path(path), true);
```
[the remark this answers is not in the document]
Comment (rgdafonso) [reply src/main/java/com/leaf/billing/utils/SparkCommands.java:24]: 🤣 🤣 🤣

**Pull request merged**